contracts/cw2981-royalties/src/lib.rs

```diff
@@ -0,0 +1,255 @@
+pub mod msg;
+pub mod query;
+
+use cosmwasm_std::to_binary;
+pub use query::{check_royalties, query_royalties_info};
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::msg::Cw2981QueryMsg;
+use cosmwasm_std::Empty;
+use cw721_base::Cw721Contract;
+pub use cw721_base::{ContractError, InstantiateMsg, MintMsg, MinterResponse};
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug, Default)]
+pub struct Trait {
+    pub display_type: Option<String>,
+    pub trait_type: String,
+    pub value: String,
+}
+
+// see: https://docs.opensea.io/docs/metadata-standards
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug, Default)]
+pub struct Metadata {
+    pub image: Option<String>,
+    pub image_data: Option<String>,
+    pub external_url: Option<String>,
+    pub description: Option<String>,
+    pub name: Option<String>,
+    pub attributes: Option<Vec<Trait>>,
+    pub background_color: Option<String>,
+    pub animation_url: Option<String>,
+    pub youtube_url: Option<String>,
+    /// specify whether royalties are set on this token
+    pub royalty_payments: bool,
+    /// This is how much the minter takes as a cut when sold
+    pub royalty_percentage: Option<u64>,
+    /// The payment address, may be different to or the same
+    /// as the minter addr
+    /// question: how do we validate this?
+    pub royalty_payment_address: Option<String>,
+}
+
+pub type Extension = Option<Metadata>;
+
+pub type MintExtension = Option<Extension>;
+
+pub type Cw2981Contract<'a> = Cw721Contract<'a, Extension, Empty>;
+pub type ExecuteMsg = cw721_base::ExecuteMsg<Extension>;
+
+#[cfg(not(feature = "library"))]
+pub mod entry {
+    use super::*;
+
+    use cosmwasm_std::entry_point;
+    use cosmwasm_std::{Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+
+    #[entry_point]
+    pub fn instantiate(
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: InstantiateMsg,
+    ) -> StdResult<Response> {
+        Cw2981Contract::default().instantiate(deps, env, info, msg)
+    }
+
+    #[entry_point]
+    pub fn execute(
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: ExecuteMsg,
+    ) -> Result<Response, ContractError> {
+        Cw2981Contract::default().execute(deps, env, info, msg)
+    }
+
+    #[entry_point]
+    pub fn query(deps: Deps, env: Env, msg: Cw2981QueryMsg) -> StdResult<Binary> {
+        match msg {
+            Cw2981QueryMsg::RoyaltyInfo {
+                token_id,
+                sale_price,
+            } => to_binary(&query_royalties_info(deps, token_id, sale_price)?),
+            Cw2981QueryMsg::CheckRoyalties {} => to_binary(&check_royalties(deps)?),
+            _ => Cw2981Contract::default()
+                .query(deps, env, msg.into())
+                .map_err(|err| err),
```

Comment:
Is this necessary? `query` should return the expected result type.

Reply:
you mean `map_err`? or the `into`?

Reply:
Yeah seems like you're right, 'map_err` not needed